### DISCUSSION TITLE
microk8s when starting consuming pod, it hangs on container creation.

### DISCUSSION THREAD

**Discussion by palmtop** [kubernetes-csi/csi-driver-smb#805] [Q&A] [answered]
I have a small microk8s cluster with 3 nodes on Ubuntu 22.04. I have installed it with helm, using the proposed --set linux.kubelet="/var/snap/microk8s/common/var/lib/kubelet" and with --set linux.dnsPolicy=ClusterFirstWithHostNet.

The installation goes without any problem, the csi pods start normally. I can create the PV and PVC without any problem, they are created and shows as bound, but when I start the pod which consumes the volume claim, it is hung at the container creation. I have tried two different pods, none of them was working and I have tweaked the PV and PVC configuration to no success.

I did the troubleshooting steps, I have one control pod and 3 node pods, but when I check their logs, I see only the startup messages, and that they are ready for receive connections, but there is no sign that they tried to mount anything.

(During the troubleshooting I discovered that cifs-util was not installed, but now I installed it in all 3 nodes, and from the nodes I am able to mount the cifs shares.)

What could be wrong?

**Comment by andyzhangx** [comment, marked as answer]
what's the error msg in pod description?

**Comment by palmtop** [comment]
Hi Andy!

Thank you for the help, actually the error was, in the pod description, that I misspelled the name of the secret containing the SMB authentication. After fixing that it started to work.
I feel myself so stupid, I looked into get pod -o yaml, I looked into the logs, looked into the csi containers, but did not look in the describe pod.
